Source/WebCore/platform/KeyboardScrollingAnimator.cpp

```diff
@@ -339,13 +341,27 @@ void KeyboardScrollingAnimator::stopKeyboardScrollAnimation()
     m_currentKeyboardScroll = std::nullopt;
 }
 
-void KeyboardScrollingAnimator::handleKeyUpEvent()
+void KeyboardScrollingAnimator::handleKeyUpEvent(Function<void()>&& callback)
 {
+    // FIXME: this fixes the seg fault, but there definitely is a better way to do this
+    auto maybeThis = this;
+    if (!maybeThis)
+        return;
     if (!m_scrollTriggeringKeyIsPressed)
         return;
 
     stopKeyboardScrollAnimation();
```

Comment:
should this be called here? or below line 357? or somewhere else entirely?

Reply:
without it, scrolling never ceases, so I put it here without knowing any other unintended consequences